framework/py/flwr/client/grpc_rere_client/connection.py

```diff
@@ -273,8 +274,9 @@ def receive() -> Optional[Message]:
         if message_proto:
             msg_id = message_proto.metadata.message_id
             run_id = message_proto.metadata.run_id
+            object_tree = response.message_object_trees[0]
             all_object_contents = pull_objects(
-                list(response.objects_to_pull[msg_id].object_ids) + [msg_id],
+                [tree.object_id for tree in iterate_object_tree(object_tree)],
```

Comment:
```suggestion
                [node.object_id for node in iterate_object_tree(object_tree)],
```

I suppose here it's more appropriate to use the term `node` (of a tree) ?